config/services/dataplane_v1beta1_openstackdataplaneservice_configurenetwork.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneService
+metadata:
+  labels:
+    app.kubernetes.io/name: openstackdataplaneservice
+    app.kubernetes.io/instance: openstackdataplaneservice-sample
+    app.kubernetes.io/part-of: dataplane-operator
+    app.kubernetes.io/managed-by: kustomize
+    app.kubernetes.io/created-by: dataplane-operator
+  name: configure-network
+spec:
+  label: dataplane-deployment-configure-network
+  role:
```

Comment:
this looks similar but not the same as how you specify a play when defining an in line playbook for the AEE operator.

is there a reason you are diverging and using roles here's nice the Hosts and stragey fields make this look more like a playbook then an ansibel role.?

Reply:
I can't make sense of what you're asking. Can you phrase it a different way?

Reply:
Can we keep this discussion moving so we can make progress?

Reply:
It's my understanding that these issues have been resolved in offline conversations. Please chime in if there's still any objections to merging this.

Reply:
what is the idea of having `services` instead of using `samples`?

Reply:
These aren't samples. They will be the actual services CR's that need to be created for the default dataplane services.

Reply:
i was refing to useing the term role here and not play.

roles in ansibel do not have a host selector or execution stragey

    name: "Deploy EDPM Network"
    hosts: "all"
    strategy: "linear"

those are set on a playbook

so the content of the role element below is actually an abstraction fo an ansibel playbook

in the AEE operator we use the key "play" to model inline playbooks
https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/blob/main/examples/openstack-ansibleee-play.yaml#L8-L14

the host key in particular I'm not sure makes sense to expose or at least require

implicitly, if you are deifing a composable service on a dataplaneNode it would only apply to that one host and similarly on a role it should only apply to that role.

I don't think we should be allowing you to run this on a subset of hosts within a role makes sense or it should not be the norm.
so if the host sub key is supported  then it should default to all and not be required

but that raise the question fo the type of this field.

here you have defined it as a mapping type but in the AEE operator its just a string so its a direct passthrough to ansible.

if you want to support Ansibel in this way accepting this as s string i think makes more sense and just define it inline.

this is not blocking by any means but i think we should be trying to keep this interface and the AEE interface consistent.

Reply:
@fultonj this issue is not adressed

Reply:
This is consistent with the Role interface from OpenStackAnsibleEEE. Which supports specifying roles based the Role struct defined in its openstack_ansibleee_types.go, and that is what I'm making use of here:

https://github.com/openstack-k8s-operators/dataplane-operator/pull/158/files#diff-be99dc960fff7094f26203ceac52ad519db728b17568a00d1bc1d9b48b811bacR34

That's also what we used in network.go (and still in other services not migrated to this interface).